### PR TITLE
Fix SchemaError when multiple computed fields point to the same type

### PR DESCRIPTION
**Pull request by adriangb**
Fixes https://github.com/pydantic/pydantic/issues/5980

Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by adriangb** [comment]
please review

File: pydantic/_internal/_core_utils.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def walk(self, schema: core_schema.CoreSchema, f: Walk) -> core_schema.CoreSchem
         return f(schema.copy(), self._walk)
 
     def _walk(self, schema: core_schema.CoreSchema, f: Walk) -> core_schema.CoreSchema:
-        ser_schema: core_schema.SerSchema | None = schema.get('serialization', None)  # type: ignore
         schema = self._schema_type_to_method[schema['type']](schema, f)
+        ser_schema: core_schema.SerSchema | None = schema.get('serialization', None)  # type: ignore
```
Comment (adriangb) [inline pydantic/_internal/_core_utils.py:254]: Ups not technically necessary in the end but this ordering is better anyway.

File: pydantic/_internal/_core_utils.py [hunk 2]
```diff
@@ -366,6 +366,13 @@ def handle_model_fields_schema(self, schema: core_schema.ModelFieldsSchema, f: W
         if 'extra_validator' in schema:
             schema['extra_validator'] = self.walk(schema['extra_validator'], f)
         replaced_fields: dict[str, core_schema.ModelField] = {}
+        replaced_computed_fields: list[core_schema.ComputedField] = []
+        for computed_field in schema.get('computed_fields', None) or ():
+            replaced_field = computed_field.copy()
+            replaced_field['return_schema'] = self.walk(computed_field['return_schema'], f)
+            replaced_computed_fields.append(replaced_field)
+        if replaced_computed_fields:
+            schema['computed_fields'] = replaced_computed_fields
```
Comment (Kludex) [inline pydantic/_internal/_core_utils.py:375]: Is `BaseModel` the only class that takes the `@computed_field`s in consideration?

**Review by Kludex — APPROVED** [review]

File: pydantic/_internal/_core_utils.py (hunk 2 above)
Comment (adriangb) [reply pydantic/_internal/_core_utils.py:375]: Good point, added tests.

**Pull request merged**